### ISSUE TITLE
Function like macro as builder alternative

### ISSUE THREAD

**Issue by Plebshot** [juhaku/utoipa#480]
First off: Amazing project. We've been looking something like this for a while.

I have a small question and a possible usage enhancement idea. I tried to use some code constants to create the api docs like so:

```rust
const SERVER_URL: &str = "https://example.com";

#[derive(OpenApi)]
#[openapi(
    servers((url = SERVER_URL))
    // ...
)]
struct ApiDoc;
```

This doesn't work due to `expected string literal`.
I know I could manually use the whole Builder syntax, or just modify the specific parts dynamically with `Modify`. But having it all in one place in a more readable format would obviously be nicer.
Perhaps, we could use a function like macro in place of the builder pattern like so:

```rust
const SERVER_URL: &str = "https://example.com";

fn generate_api() {
        let docs = openapi!(
            servers((url = SERVER_URL)
            // ...
        ));
        // ... do some other things with the docs
} 
```

I'm wondering, would it be a lot of work to make this work?

**Comment by juhaku** [comment]
Hey @Plebshot ,

> I'm wondering, would it be a lot of work to make this work?

In general having support for function macro should not be a very large contribution if the same parsing logic can be used as for the derive macro.  Otherwise it is entirely a seprate functionality. In such a case the effort would be a lot bigger. Nonetheless adding support for the derive macro to handle expressions as literal strings would invalidate the reason for the function macro (at least that reason). Extending the `OpenApi` and other derive macros to handle the case above should be doable with quite small effort and can be added in the future. 
> ```rust
> const SERVER_URL: &str = "https://example.com";
> 
> #[derive(OpenApi)]
> #[openapi(
>    servers((url = SERVER_URL))
>    // ...
> )]
> struct ApiDoc;
> ```

**Comment by Drifter-J** [comment]
utoipa already have `LitStrOrExpr` support for servers(url), operation_id, and etc. This can be closed @juhaku

**Comment by juhaku** [comment]
> utoipa already have `LitStrOrExpr` support for servers(url), operation_id, and etc. This can be closed [@juhaku](https://github.com/juhaku)

True, closing, `LitStrOrExpr` solves the underlying issue. I don't see how we need yet another style of API to do the same thing. Just adds maintenance overhead.
